app/src/lib/helpers/regex.ts

```diff
@@ -51,3 +51,19 @@ export function getMatches(text: string, re: RegExp): Array<RegExpExecArray> {
 export function escapeRegExp(expression: string) {
   return expression.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&')
 }
+
+export function getFileFromExceedsError(error: string): string | null {
+  const endRegex = /(\sis\s\d+.\d+ [a-zA-Z]+;\sthis\sexceeds\sGitHub's\sfile\ssize\slimit\sof\s100.00\sMB)/gm
+  const beginRegex = /(^remote:\serror:\sFile\s)/gm
+  const beginMatch = beginRegex.exec(error)
+  const endMatch = endRegex.exec(error)
+
+  if (beginMatch && beginMatch.index && endMatch && endMatch.index) {
```

Comment:
FYI nowadays we prefer specific checks for `null` / `undefined`:
```suggestion
  if (beginMatch !== null && beginMatch.index && endMatch !== null && endMatch.index) {
```

What are the `.index` checks for exactly? 🤔

Reply:
I just really like conditional statements... 

I was using `matches` before instead of `exec` in which the index might be undefined. I'll update it while I make the other changes.